### PR TITLE
Correcting bugs in SMT 2.0 

### PR DESCRIPTION
**Pull request by Paul-Saves**
Finished and okay for merging

### CODE REVIEW

File: smt/surrogate_models/krg_based.py [hunk 1]
```diff
@@ -1159,7 +1159,7 @@ def _predict_values(self, x: np.ndarray, is_acting=None) -> np.ndarray:
             Evaluation point output variable values
         """
         # Initialization
-        if is_acting is None:
+        if is_acting is not None or not (self.design_space.is_all_cont):
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:1162]: This is weird because you never use the `is_acting` array content given as argument.

File: smt/surrogate_models/krg_based.py [hunk 2]
```diff
@@ -1319,7 +1319,7 @@ def _predict_variances(self, x: np.ndarray, is_acting=None) -> np.ndarray:
             Evaluation point output variable MSE
         """
         # Initialization
-        if is_acting is None:
+        if is_acting is not None or not (self.design_space.is_all_cont):
```
Comment (relf) [inline smt/surrogate_models/krg_based.py:1322]: Same thing here.

**Comment by relf** [comment]
I think you should add some tests (is_acting being None or something)  to reveal the bug then make them pass with the relevant fix.

File: smt/surrogate_models/krg_based.py (hunk 1 above)
Comment (Paul-Saves) [reply smt/surrogate_models/krg_based.py:1162]: It is used by gower_componentwise_distances

Comment (relf) [reply smt/surrogate_models/krg_based.py:1162]: I mean you have changed the logic. Before your fix when `is_acting` was None  we retrieve it from ` self.design_space.correct_get_acting(x)`, otherwise we just use it.

Now if we pass an `is_acting` array, you seem to override it with `x, is_acting = self.design_space.correct_get_acting(x)`. So whether or not it is right, the intent is not clear and it looks a bit confusing.

Moreover as all the tests were successful before I think we should add some tests to cover the logic we want and document the expected behaviour as well.
